cogs/tas2gif.py

```diff
@@ -0,0 +1,131 @@
+from discord.ext import commands
+from discord.ext import tasks
+import discord
+import requests
+from config import celia_home
+import asyncio, aiohttp
+import os
+import subprocess, shutil
+
+gif_processing = False
+
+class tas2gif(commands.Cog):
+
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.command()
+    async def tas2gif(self, ctx, *args):
+
+        global gif_processing
+        if (gif_processing): 
+            await ctx.send('A GIF is currently processing, try again later.')
+            return 
+        gif_processing = True
+
+        if len(args) not in (2, 3):
+            await tas2gif_error(f'Usage: !tas2gif <cartname> <level_index> [url]', ctx)
+            return
+
+        cartname = args[0]
+        level = args[1]
+
+        # Download file
+        attachment_url =  args[2] if len(args) > 2  else ctx.message.attachments[0].url
+        try:
+            attachment_file = await fetch(attachment_url)
+        except ConnectionError as e:
+            await tas2gif_error(f"Error while getting TAS file (HTTP {e}).", ctx)
+            return
+        except:
+            await tas2gif_error("Could not download TAS file!", ctx)
+            return       
+
+
+        # Write to disk
+        tas_file = "files/" + "recordme_" + cartname + "_" + level + ".tas"      
+        with open(tas_file, "w+b") as f:
+            f.write(attachment_file)
+
+        # Estimate length of TAS file    
+        try:
+            total_frames = attachment_file.decode("utf-8").count(",")
```

Comment:
Balloon seeding (the stuff in `[]`) will break this, as it also contains commas

Reply:
Check out the process_inputs function in tas.py since it does something similar